porcupine/_state.py

```diff
@@ -125,12 +125,11 @@ def quit() -> None:
     with :meth:`~porcupine.tabs.TabManager.close_tab` and widgets are
     destroyed.
     """
-    for tab in get_tab_manager().tabs():
+    get_main_window().event_generate("<<PorcupineQuit>>")  # TODO: still needed?
+
+    for tab in reversed(get_tab_manager().tabs()):
```

Comment:
Idk, it just makes more sense to me to start closing the tabs from the end of the list.